File: frontend/src/api/score.ts
import { apiClient } from "@/api/client";
import type { IntakeAnswers } from "@/types/intake";
import type { ScoreBreakdown } from "@/types/score";

export function calculateScore(intake: IntakeAnswers): Promise<ScoreBreakdown> {
  return apiClient.calculateScore(intake);
}

File: backend/app/routers/score.py
from fastapi import APIRouter

from app.ml.factor_engine import compute_factor_scores
from app.schemas.score import IntakeAnswers, ScoreBreakdown

router = APIRouter(prefix="/api/score", tags=["score"])


@router.post("/calculate", response_model=ScoreBreakdown)
def calculate_score(payload: IntakeAnswers) -> ScoreBreakdown:
    return compute_factor_scores(payload)
